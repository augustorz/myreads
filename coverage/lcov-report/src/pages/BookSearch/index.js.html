<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/pages/BookSearch/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/pages/BookSearch</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/13</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import * as BooksAPI from '../../api/BooksAPI';
import './index.css';
&nbsp;
import BooksGrid from '../../components/BooksGrid';
&nbsp;
class BookSearch extends Component {
  state = {
    query: '',
    books: [],
  }
&nbsp;
  handleSearch = ({ target: { value: query } }) =&gt; {
    this.setState({
      books: [],
      query,
    });
&nbsp;
    if (query) {
      this.searchBooks();
    }
  }
&nbsp;
  searchBooks = async () =&gt; {
    const { query } = this.state;
&nbsp;
    try {
      const books = await BooksAPI.search(query);
&nbsp;
      if (books &amp;&amp; !books.error) {
        this.setState({ books });
      }
    } catch (error) {
      throw error;
    }
  }
&nbsp;
  render() {
    const { query, books } = this.state;
    const { onChangeBookShelf } = this.props;
&nbsp;
    return (
      &lt;div className="search-books"&gt;
        &lt;div className="search-books-bar"&gt;
          &lt;Link className="close-search" to="/"&gt;Close&lt;/Link&gt;
          &lt;div className="search-books-input-wrapper"&gt;
            &lt;input
              type="text"
              placeholder="Search by title or author"
              value={query}
              onChange={this.handleSearch}
            /&gt;
          &lt;/div&gt;
        &lt;/div&gt;
        &lt;div className="search-books-results"&gt;
          &lt;BooksGrid books={books} onChangeBookShelf={onChangeBookShelf} /&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
BookSearch.propTypes = {
  onChangeBookShelf: PropTypes.func.isRequired,
};
&nbsp;
export default BookSearch;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 14 2019 08:26:49 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
