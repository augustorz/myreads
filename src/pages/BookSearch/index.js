import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import * as BooksAPI from '../../BooksAPI';

import Book from '../../components/Book';

class BookSearch extends Component {
  state = {
    query: '',
    books: [],
  }

  handleSearch = (event) => {
    const query = event.target.value;

    this.setState({
      books: [],
      query,
    });

    if (query) {
      this.searchBooks();
    }
  }

  searchBooks = async () => {
    const { query } = this.state;

    try {
      const books = await BooksAPI.search(query);

      if (books && !books.error) {
        this.setState({ books });
      }
    } catch (error) {
      throw error;
    }
  }

  render() {
    const { query, books } = this.state;
    const { onChangeBookShelf } = this.props;

    return (
      <div className="search-books">
        <div className="search-books-bar">
          <Link className="close-search" to="/">Close</Link>
          <div className="search-books-input-wrapper">
            <input
              type="text"
              placeholder="Search by title or author"
              value={query}
              onChange={this.handleSearch}
            />
          </div>
        </div>
        <div className="search-books-results">
          <ol className="books-grid">
            {books.map(book => (
              <Book key={book.id} book={book} onChangeBookShelf={onChangeBookShelf} />
            ))}
          </ol>
        </div>
      </div>
    );
  }
}

BookSearch.propTypes = {
  onChangeBookShelf: PropTypes.func.isRequired,
};

export default BookSearch;
