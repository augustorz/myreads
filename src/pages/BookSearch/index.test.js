import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import createBooks from '../../mocks/books';

import BookSearch from '.';

describe('BookSearch', () => {
  let routerWrapper;
  let wrapper;

  beforeEach(() => {
    routerWrapper = mount(
      <MemoryRouter>
        <BookSearch
          onChangeBookShelf={jest.fn()}
        />
      </MemoryRouter>,
    );

    wrapper = routerWrapper.find(BookSearch);
  });

  it('should match snapshot', () => {
    expect(routerWrapper.html()).toMatchSnapshot();
  });

  describe('handleSearch', () => {
    let event;

    describe('when query is empty', () => {
      beforeEach(() => {
        event = {
          target: {
            value: '',
          },
        };

        wrapper.instance().searchBooks = jest.fn();
        wrapper.instance().handleSearch(event);
      });

      it('should set books state as an empty array', () => {
        expect(wrapper.state('books')).toEqual([]);
      });

      it('should set query state as an empty string', () => {
        expect(wrapper.state('query')).toEqual('');
      });

      it('should not call searchBooks', () => {
        expect(wrapper.instance().searchBooks).not.toBeCalled();
      });
    });

    describe('when query is present', () => {
      beforeEach(() => {
        event = {
          target: {
            value: 'Search Query',
          },
        };

        wrapper.instance().searchBooks = jest.fn();
        wrapper.instance().handleSearch(event);
      });

      it('should set books state as an empty array', () => {
        expect(wrapper.state('books')).toEqual([]);
      });

      it('should set query state as event value', () => {
        expect(wrapper.state('query')).toEqual(event.target.value);
      });

      it('should call searchBooks', () => {
        expect(wrapper.instance().searchBooks).toBeCalled();
      });
    });

    describe('searchBooks', () => {
      beforeEach(() => {
        fetch.resetMocks();
      });

      describe('if response is empty', () => {
        beforeEach(async (done) => {
          wrapper.setState = jest.fn();

          fetch.mockResponse(JSON.stringify({}));

          await wrapper.instance().searchBooks();
          done();
        });

        it('should not call setState', () => {
          expect(wrapper.setState).not.toBeCalled();
        });
      });

      describe('if response is present and has no errors', () => {
        let books;

        beforeEach(async (done) => {
          books = createBooks({ quantity: 2 });

          fetch.mockResponse(JSON.stringify({
            books,
          }));

          await wrapper.instance().searchBooks();
          done();
        });

        it('should set books state as response data', () => {
          expect(wrapper.state('books')).toEqual(books);
        });
      });

      describe('if response is present but has errors', () => {
        beforeEach(async (done) => {
          wrapper.setState = jest.fn();

          fetch.mockResponse(JSON.stringify({
            books: {
              error: {},
            },
          }));

          await wrapper.instance().searchBooks();
          done();
        });

        it('should not call setState', () => {
          expect(wrapper.setState).not.toBeCalled();
        });
      });
    });
  });
});
