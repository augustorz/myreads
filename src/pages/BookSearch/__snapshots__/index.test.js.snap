// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookSearch should match snapshot 1`] = `"<div class=\\"search-books\\"><div class=\\"search-books-bar\\"><a class=\\"close-search\\" href=\\"/\\">Close</a><div class=\\"search-books-input-wrapper\\"><input type=\\"text\\" placeholder=\\"Search by title or author\\" value=\\"\\"></div></div><div class=\\"search-books-results\\"><ol class=\\"books-grid\\"></ol></div></div>"`;
